### PR TITLE
Quality of life `smart validate` Improvements

### PR DESCRIPTION
**Pull request by MattToast**
Quality of life `smart validate` improvements:
- Set `CUDA_VISIBLE_DEVICES` environment variable within `smart validate` prior to importing any ML deps to prevent false negatives on multi-GPU systems
- Move SmartRedis logs from standard out to dedicated log file in the validation temporary directory
- Suppress `sklearn` deprecation warning by pinning `KMeans` constructor argument

### CODE REVIEW

**Review by al-rigazzi — APPROVED** [review]
LGTM, I love quality of life improvements!

**Review by ashao — APPROVED** [review]
Looks good! Thanks for finding this

File: smartsim/_core/_cli/validate.py [hunk 1]
```diff
@@ -90,7 +89,16 @@ def execute(
     """
     backends = installed_redisai_backends()
     try:
-        with _VerificationTempDir(dir=os.getcwd()) as temp_dir:
+        with contextlib.ExitStack() as ctx:
+            temp_dir = ctx.enter_context(_VerificationTempDir(dir=os.getcwd()))
+            validate_env = {
+                "CUDA_VISIBLE_DEVICES": "0",
```
[the remark this answers is not in the document]
Comment (MattToast) [reply smartsim/_core/_cli/validate.py:95]: Good call! Will add!!

File: smartsim/_core/_cli/validate.py [hunk 2]
```diff
@@ -155,12 +157,12 @@ def test_install(
         logger.info("Verifying Tensor Transfer")
         client.put_tensor("plain-tensor", np.ones((1, 1, 3, 3)))
         client.get_tensor("plain-tensor")
-        if with_tf:
-            logger.info("Verifying TensorFlow Backend")
-            _test_tf_install(client, location, device)
         if with_pt:
             logger.info("Verifying Torch Backend")
             _test_torch_install(client, device)
+        if with_tf:
```
Comment (ashao) [inline smartsim/_core/_cli/validate.py:163]: If this fixes the problem we found, could you move the Tensorflow test to the very end?

Comment (MattToast) [reply smartsim/_core/_cli/validate.py:163]: absolutely!!

**Pull request merged**